app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java

```diff
@@ -173,166 +201,171 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
     }
 
     public void handleRegisterUser() {
-        fAuth = FirebaseAuth.getInstance();
-        fStore = FirebaseFirestore.getInstance();
-        if (fAuth.getCurrentUser() != null) {
-            userId = fAuth.getCurrentUser().getUid();
+        if (userId != null) {
             TextView[] textViewArray = {fullName, email, phone};
             FirebaseInteraction.retrieveDataFromFirebase(fStore, userId, textViewArray, ActivityDescriptionActivity.this);
         }
     }
 
     public void logout(View view) {
-        if (fAuth.getCurrentUser() != null) {
-            FirebaseAuth.getInstance().signOut(); // this will do the logout of the user from Firebase
+        if (userId != null) {
+            fAuth.signOut(); // this will do the logout of the user from Firebase
             startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
             finish();
         }
     }
 
 
+    /**
+     * Title of the activity
+     */
     private void createTitleView() {
-        // Title of the activity
         TextView activityTitle = (TextView) findViewById(R.id.activity_title_description);
-        if (act != null) activityTitle.setText(act.getTitle());
+        activityTitle.setText(activity.getTitle());
     }
 
+    /**
+     * Number of participants of the activity
+     */
     private void createParticipantNumberView() {
-        // Number of participants of the activity
         numberParticipantsView = (TextView) findViewById(R.id.activity_number_description);
         participantNamesView = (TextView) findViewById(R.id.activity_participants_description);
-        if (act != null) {
```

Comment:
Why did you always remove this check "act!=null" ?

Reply:
Because I have already check if the activity is not null before call this function

Reply:
Ok ok, my bad, all good!